### PR TITLE
argparse: fix error __init__, add BooleanOptionalAction for py39

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

File: stdlib/2and3/argparse.pyi [hunk 1]
```diff
@@ -283,6 +284,21 @@ class Action(_AttributeHolder):
     if sys.version_info >= (3, 9):
         def format_usage(self) -> str: ...
 
+if sys.version_info >= (3, 9):
+    class BooleanOptionalAction(Action):
+        def __init__(
+            self,
+            option_strings: Sequence[Text],
```
Comment (JelleZijlstra) [inline stdlib/2and3/argparse.pyi:291]: This is Python 3-only so no real need to use `Text`

File: stdlib/2and3/argparse.pyi [hunk 2]
```diff
@@ -283,6 +284,21 @@ class Action(_AttributeHolder):
...
+            dest: Text,
+            const: Any = ...,
+            default: Any = ...,
+            type: Optional[Union[Callable[[Text], _T], Callable[[str], _T], FileType]] = ...,
```
Comment (JelleZijlstra) [inline stdlib/2and3/argparse.pyi:295]: Should this class be generic over _T?

File: stdlib/2and3/argparse.pyi [hunk 3]
```diff
@@ -283,6 +284,21 @@ class Action(_AttributeHolder):
     if sys.version_info >= (3, 9):
         def format_usage(self) -> str: ...
 
+if sys.version_info >= (3, 9):
+    class BooleanOptionalAction(Action):
+        def __init__(
+            self,
+            option_strings: Sequence[Text],
+            dest: Text,
+            const: Any = ...,
```
Comment (JelleZijlstra) [inline stdlib/2and3/argparse.pyi:293]: I don't fully understand the description of this argument but it sounds like it and default should be of type `Optional[_T]`.

File: stdlib/2and3/argparse.pyi (hunk 2 above)
Comment (hauntsaninja) [reply stdlib/2and3/argparse.pyi:295]: I don't think so? Not sure what that would allows us to accomplish

**Comment by hauntsaninja** [comment]
Re: the unused in implementation, we'll see what happens https://github.com/python/cpython/pull/11478#pullrequestreview-423968410

**Pull request merged**